nolearn/lasagne/handlers.py

```diff
@@ -92,3 +90,78 @@ def __call__(self, nn, train_history):
                 pickle.dump(nn, f, -1)
         else:
             nn.save_params_to(path)
+
+
+class PrintLayerInfo:
+    def __init__(self):
+        pass
+
+    def __call__(self, nn):
+        message = self._get_greeting(nn)
+        print(message)
+        print("## Layer information")
+        print("")
+
+        layers_contain_conv2d = is_conv2d(nn.layers_.values())
```

Comment:
For Python 3, if you expect to pass a list, then use `list(nn.layers_.values())`.  Otherwise what you get is a view that's _only_ iterable.